### DISCUSSION TITLE
Depends doesn't work with dependencies=[Depends(something)]

### DISCUSSION THREAD

**Discussion by Enola-guy** [fastapi/fastapi#7832] [Questions] [answered]
```python
dependencies=[Depends(logged_in)]
``` 
doesn't have the same behavior as  :  
```python
login: str = Depends(logged_in)
``` 

You can run this code as is :  

```python
from fastapi import FastAPI
from fastapi.security import HTTPBasic, HTTPBasicCredentials

api = FastAPI()
security = HTTPBasic()

def logged_in(credentials: HTTPBasicCredentials = Depends(security)):
    if credentials.username == "toto":
        return "good"
    else:
        raise HTTPException(
        status_code=HTTP_401_UNAUTHORIZED,
        detail="Incorrect email or password",
        headers={"WWW-Authenticate": "Basic"},
    )

@api.get("/")
async def homepage(dependencies=[Depends(logged_in)]):
    return {"no": "authentication"}


@api.get("/log")
async def homepage(login: str = Depends(logged_in)):
    return {"log": "working"}
```

I would expect both routes `/log` and `/` to ask for authentication but only `/log` does it

**env**
Python 3.7
FastAPI 0.44.0

I hope my issue is clean this is my first ever on an open source project :sweat_smile: 

By the way thanks for your great work.

**Comment by dmontagu** [comment, marked as answer]
The `dependencies=[Depends(loggin_in)]` actually needs to go into the route decorator, not the endpoint signature. So it should look like this:

```python
@api.get("/", dependencies=[Depends(logged_in)])
async def homepage():
    return {"no": "authentication"}
```

Currently, the only introspection FastAPI does to look for dependencies in the endpoint signature is to check if the default value is of type `Depends` (or `Security`). So providing a list of `Depends` like you've done won't work. If you want to achieve something similar, you should just create a separate dependency that returns the list you want.

Hopefully this helps! Let us know if you have any more confusion or this doesn't answer your question.

------

> I hope my issue is clean this is my first ever on an open source project 😅

This is a fine issue, thanks for contributing!

**Comment by Enola-guy** [comment]
Ok I've checked again and the doc is totally clear, I got confused, my bad. Thanks for the answer.
If someone is wondering where it's written...
**Right here:**  
https://fastapi.tiangolo.com/tutorial/dependencies/dependencies-in-path-operation-decorators/

**Comment by tiangolo** [comment]
Thanks for the help here @dmontagu ! :clap: :bow: 

Thanks for reporting back and closing the issue @Enola-guy  :+1: And especial thanks for adding the link to the docs :nerd_face: :cake:
